daemon/emer-daemon.c

```diff
@@ -988,7 +1016,7 @@ upload_events (EmerDaemon         *self,
   EmerDaemonPrivate *priv = emer_daemon_get_instance_private (self);
 
   GTask *upload_task =
-    g_task_new (self, NULL /* GCancellable */, callback, user_data);
+    g_task_new (self, g_cancellable_new (), callback, user_data);
```

Comment:
Are you not leaking a reference to the cancellable here? `g_task_new()` will take an extra ref to it.